projects/graph/social/social.py

```diff
@@ -1,72 +1,83 @@
+import random
 
 
 class User:
-    def __init__(self, name):
-        self.name = name
+  def __init__(self, name):
+    self.name = name
+
 
 class SocialGraph:
-    def __init__(self):
-        self.lastID = 0
-        self.users = {}
-        self.friendships = {}
-
-    def addFriendship(self, userID, friendID):
-        """
-        Creates a bi-directional friendship
-        """
-        if userID == friendID:
-            print("WARNING: You cannot be friends with yourself")
-        elif friendID in self.friendships[userID] or userID in self.friendships[friendID]:
-            print("WARNING: Friendship already exists")
-        else:
-            self.friendships[userID].add(friendID)
-            self.friendships[friendID].add(userID)
-
-    def addUser(self, name):
-        """
-        Create a new user with a sequential integer ID
-        """
-        self.lastID += 1  # automatically increment the ID to assign the new user
-        self.users[self.lastID] = User(name)
-        self.friendships[self.lastID] = set()
-
-    def populateGraph(self, numUsers, avgFriendships):
-        """
-        Takes a number of users and an average number of friendships
-        as arguments
-
-        Creates that number of users and a randomly distributed friendships
-        between those users.
-
-        The number of users must be greater than the average number of friendships.
-        """
-        # Reset graph
-        self.lastID = 0
-        self.users = {}
-        self.friendships = {}
-        # !!!! IMPLEMENT ME
-
-        # Add users
-
-        # Create friendships
-
-    def getAllSocialPaths(self, userID):
-        """
-        Takes a user's userID as an argument
-
-        Returns a dictionary containing every user in that user's
-        extended network with the shortest friendship path between them.
-
-        The key is the friend's ID and the value is the path.
-        """
-        visited = {}  # Note that this is a dictionary, not a set
-        # !!!! IMPLEMENT ME
-        return visited
+  def __init__(self):
+    self.lastID = 0
+    self.users = {}
+    self.friendships = {}
+
+  def addFriendship(self, userID, friendID):
```

Comment:
Now this is clean!, great job of converting your code to 2 instead of 4, great job!!